src/main/java/graphql/ProfilerResult.java

```diff
@@ -310,11 +310,21 @@ public Map<String, Object> shortSummaryMap() {
                 Assert.assertShouldNeverHappen();
             }
         }
-        result.put("dataFetcherResultTypes", Map.of(
-                DataFetcherResultType.COMPLETABLE_FUTURE_COMPLETED.name(), "(count:" + completedCount + ", invocations:" + completedInvokeCount + ")",
-                DataFetcherResultType.COMPLETABLE_FUTURE_NOT_COMPLETED.name(), "(count:" + notCompletedCount + ", invocations:" + notCompletedInvokeCount + ")",
-                DataFetcherResultType.MATERIALIZED.name(), "(count:" + materializedCount + ", invocations:" + materializedInvokeCount + ")"
-        ));
+        LinkedHashMap<String, Object> dFRTinfo = new LinkedHashMap<>(3);
+        LinkedHashMap<String, Integer> completableFutureCompleted = new LinkedHashMap<>(2);
+        completableFutureCompleted.put("count", completedCount);
+        completableFutureCompleted.put("invocations", completedInvokeCount);
+        LinkedHashMap<String, Integer> completableFutureNotCompleted = new LinkedHashMap<>(2);
+        completableFutureNotCompleted.put("count", notCompletedCount);
+        completableFutureNotCompleted.put("invocations", notCompletedInvokeCount);
+        LinkedHashMap<String, Integer> materialized = new LinkedHashMap<>(2);
+        materialized.put("count", materializedCount);
+        materialized.put("invocations", materializedInvokeCount);
+
+        dFRTinfo.put(DataFetcherResultType.COMPLETABLE_FUTURE_COMPLETED.name(), completableFutureCompleted);
+        dFRTinfo.put(DataFetcherResultType.COMPLETABLE_FUTURE_NOT_COMPLETED.name(), completableFutureNotCompleted);
+        dFRTinfo.put(DataFetcherResultType.MATERIALIZED.name(), materialized);
+        result.put("dataFetcherResultTypes", dFRTinfo);
```

Comment:
I know, it's much more verbose this way compared to Map.of

I plan to ban Map.of in the near future https://github.com/graphql-java/graphql-java/issues/3989 - in another place, it was important to have consistent ordering when serialising source location. It doesn't matter as much here but might as well make it consistent and ban Map.of

Reply:
it is the right thing todo ... we don't want Map.of. 

We should invent a version of it that is based on LinkedHashMap

Reply:
I made that a todo on the ban Map.of ticket